File: crates/auto-lang/test/a2r/035_inheritance/inheritance.expected.rs
struct Animal {
    name: String,
}

impl Animal {
    fn speak(&self) {
        println!("Animal sound");
    }
}

struct Dog {
    name: String,
    breed: String,
}

impl Dog {
    fn bark(&self) {
        println!("Woof!");
    }
    fn speak(&self) {
        println!("Animal sound");
    }
}

fn main() {
    let dog: Dog = Dog {};
    dog.name = "Buddy";
    dog.breed = "Labrador";

    dog.speak();
    dog.bark();
}

File: crates/auto-lang/test/a2c/128_inheritance/inheritance.expected.c
#include "inheritance.h"

void Animal_Speak(struct Animal *self) {
    printf("%s\n", "Animal sound");
}

void Dog_Bark(struct Dog *self) {
    printf("%s\n", "Woof!");
}
void Dog_Speak(struct Dog *self) {
    printf("%s\n", "Animal sound");
}

int main(void) {
    unknown dog = Dog();
    dog.name = "Buddy";
    dog.breed = "Labrador";

    dog.speak();
    dog.bark();

    printf("%s\n", "Test passed");
    return 0;
}
